lib/Controller/CallController.php

```diff
@@ -158,7 +161,26 @@ public function downloadParticipantsForCall(string $format = 'csv'): DataDownloa
 
 		fseek($output, 0);
 
-		return new DataDownloadResponse(stream_get_contents($output), 'participants.csv', 'text/csv');
+		// Clean the room name
+		$cleanedRoomName = preg_replace('/[\/\\:*?"<>|\- ]+/', '-', $this->room->getName());
+		// Limit to a reasonable length
+		$cleanedRoomName = substr($cleanedRoomName, 0, 100);
+
+		$timezone = 'UTC';
+		if ($this->participant->getAttendee()->getActorType() === Attendee::ACTOR_USERS) {
+			$timezone = $this->serverConfig->getUserValue($this->participant->getAttendee()->getActorId(), 'core', 'timezone', 'UTC');
+		}
+
+		try {
+			$dateTimeZone = new \DateTimeZone($timezone);
+		} catch (\DateInvalidTimeZoneException) {
+			$dateTimeZone = null;
+		}
+
+		$date = $this->timeFactory->getDateTime('now', $dateTimeZone)->format('Y-m-d');
```

Comment:
```suggestion
		$date = $this->timeFactory->getDateTime('now', $dateTimeZone)->format('Y-m-d H-i');
```

?

Reply:
Unlikely someone would have a call in the same conversation with different participants twice the same day? Even so, they could rename it later themselves

Reply:
but it will be nice to have, to know when you took the snapshot of participants (which helps a lot for student attendance timing).

Reply:
Then it will come to decision-making (e.g. students claim they've attended the call), timestamp of file creation will still be more valid than a file name